src/Slider.jsx

```diff
@@ -337,16 +337,20 @@ class Slider extends React.Component {
     const lowerOffset = this.calcOffset(lowerBound);
 
     const handleClassName = prefixCls + '-handle';
+
+    const upperClassName = handleClassName + ' ' + handleClassName + '-upper';
+    const lowerClassName = handleClassName + ' ' + handleClassName + '-lower';
+
     const isNoTip = (step === null) || (tipFormatter === null);
 
-    const upper = cloneElement(customHandle, { className: handleClassName,
-        noTip: isNoTip, tipTransitionName: tipTransitionName, tipFormatter: tipFormatter,
+    const upper = cloneElement(customHandle, { className: upperClassName,
+        noTip: isNoTip, tipTransitionName: tipTransitionName, tipFormatter: tipFormatter, upperHandle: true, lowerHandle: false,
```

Comment:
Removed them. You might still consider it though, as it makes identifying an upper/lower handle a little easier from within a custom handle component. Otherwise you'll have to search the className string for 'upper' or 'lower'.